ls1intum/Artemis
# `Exam mode`: Make participation status change only available after working time is over

### Checklist
#### General



#### Client

### Description

As noticed in https://github.com/ls1intum/Artemis/pull/7267/files#r1336404731 when trying to change the student exam to submitted while the exam is still running two error
messages are displayed that do not describe the actual problem _(= the submitted state of a student exam cannot be changed while it is in progress)_.

The issue was, that individual working times of students have not been considered for displaying the button.

This PR
* prevents that this error can occur by only making the button available when the exam is finished _(considering the individual working time)_
* introduces the usage of ngElse to reduce error-prone code
* improves the UI by indenting all categories with the same margin left
* show the end of the working time in case an individual working time was set

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exam with at least 1 registered student

1. Open the Student exam for the registered student while the exam is running
2. See that `Change Student Exam to Submitted` is not available
3. Wait until the exam is finished _(you can adjust the end date)_
4. See that `Change Student Exam to Submitted` is not available
5. Submit & unsubmit the exam, verify that the translations in the modal are displayed properly
6. Give the student a time extension, which leads to the exam not being finished => press save! 
7. See that the individual exam end date is previewed
8. See that `Change Student Exam to Submitted` is not available anymore

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-------------------:|
| [student-exam-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5555/latest/artifact/shared/Coverage-Report-Client-Tests/app/exam/manage/student-exams/student-exam-detail.component.ts.html) | 92.92% | ✅ |


### Screenshots


| Before                                                                                                                                                                                 | After                                                                                                                                                                                                                                                                                                                        |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| <img width="1248" alt="not indented consistently and without preview of exam end time" src="https://github.com/ls1intum/Artemis/assets/63976129/8a478ae8-5ae8-4508-b99f-8199247c3035"> | <img width="1248" alt="indented and with individual working time preview" src="https://github.com/ls1intum/Artemis/assets/63976129/a204e849-90e4-42d0-8a89-0bbee757a126"> <img width="1248" alt="exam over button displayed" src="https://github.com/ls1intum/Artemis/assets/63976129/f23e28e6-4b84-4351-8901-70ef0e0b1094"> |
